server/channels/app/report.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"bytes"
+	"encoding/csv"
+	"fmt"
+	"net/http"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+func (a *App) SaveReportChunk(format string, prefix string, count int, reportData []model.ReportableObject) *model.AppError {
+	switch format {
+	case "csv":
+		return a.saveCSVChunk(prefix, count, reportData)
+	}
+	return model.NewAppError("SaveReportChunk", "", nil, "unsupported report format", http.StatusInternalServerError)
+}
+
+func (a *App) saveCSVChunk(prefix string, count int, reportData []model.ReportableObject) *model.AppError {
+	var buf bytes.Buffer
+	w := csv.NewWriter(&buf)
+
+	for _, report := range reportData {
+		err := w.Write(report.ToReport())
+		if err != nil {
+			return model.NewAppError("saveCSVChunk", "", nil, "failed to write report data to CSV", http.StatusInternalServerError)
+		}
+	}
+
+	w.Flush()
+	_, appErr := a.WriteFile(&buf, makeFilename(prefix, count, "csv"))
+	if appErr != nil {
+		return appErr
+	}
+
+	return nil
+}
+
+func (a *App) CompileReportChunks(format string, prefix string, numberOfChunks int, headers []string) (string, *model.AppError) {
```

Comment:
Once the job has batched everything, upon completion we should merge all the files together. 
For a CSV, this just means appending everything into one file.